dateutil/tz/_factories.py

```diff
@@ -1,5 +1,10 @@
 from datetime import timedelta
 
+try:
+    import weakref
```

Comment:
I would change this to:

```python
try:
    from weakref import WeakRefDictionary
except ImportError:
    WeakRefDictionary = dict
```

Reply:
Actually it turns out that Python 2 has weak references, so we can just drop the whole compatibility shim. My bad.